src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -1301,13 +1301,13 @@ public function createDatabase(string $name, array $options = []): void
 
         if (isset($options['collation'])) {
             $this->execute(sprintf(
-                'CREATE DATABASE `%s` DEFAULT CHARACTER SET `%s` COLLATE `%s`',
-                $name,
+                'CREATE DATABASE %s DEFAULT CHARACTER SET `%s` COLLATE `%s`',
+                $this->quoteColumnName($name),
```

Comment:
The upstream patch used `$this->quoteTableName` which does an escape around any `.` characters in addition to calling `quoteColumnName` where it's assuming that it's a separator between schema and table name. Given that database names don't have schema references, better I think to use `quoteColumnName` directly.